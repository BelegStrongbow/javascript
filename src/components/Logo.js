import React from "react";
import styled from "styled-components";
import logo from "../images/logo.svg";

export const Logo = styled.div`
	height: ${props => props.divHeight};
	width: ${props => props.divWidth};
	background: url( ${logo} ) center center no-repeat;
	background-size: ${props => props.logoSize};
`;

Logo.propTypes = {
	logoSize: React.PropTypes.string,
	divWidth: React.PropTypes.string,
	divHeight: React.PropTypes.string,
};

Logo.defaultProps = {
	logoSize: "175px",
	divHeight: "90px",
	divWidth: "300px",
};

export default Logo;
