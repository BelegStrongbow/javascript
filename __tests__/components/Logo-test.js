import React from 'react';
import renderer from 'react-test-renderer';
import Logo from '../../src/components/Logo';

test('the logo matches the snapshot', () => {
    const component = renderer.create(
        <Logo width={"150px"} height={"60px"} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
