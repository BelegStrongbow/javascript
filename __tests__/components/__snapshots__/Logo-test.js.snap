exports[`test the logo matches the snapshot 1`] = `
<a
  href="/"
  onClick={[Function]}>
  <img
    alt="My logo alt"
    className="kqACYZ"
    size="300px"
    src="logo.svg" />
</a>
`;
